sklearnex/neighbors/common.py

```diff
@@ -264,3 +266,41 @@ def _onedal_gpu_supported(self, method_name, *data):
 
     def _onedal_cpu_supported(self, method_name, *data):
         return self._onedal_supported("cpu", method_name, *data)
+
+    def kneighbors_graph(self, X=None, n_neighbors=None, mode="connectivity"):
+        check_is_fitted(self)
+        if n_neighbors is None:
+            n_neighbors = self.n_neighbors
+
+        # check the input only in self.kneighbors
+
+        # construct CSR matrix representation of the k-NN graph
+        if mode == "connectivity":
+            A_ind = self.kneighbors(X, n_neighbors, return_distance=False)
+            xp, _ = get_namespace(A_ind)
```

Comment:
In case of Array API style coding, I think it make sense early call at once get namespace for the input.
Wouldn't be it better to move on line 272 get namespace for input X.

Reply:
What is difficult here is that X is a kwarg with a default value of None.  In that case, it would depend on array type of the stored values in onedal_estimator: https://github.com/intel/scikit-learn-intelex/blob/main/onedal/neighbors/neighbors.py#L298  It is currently numpy arrays, but that's not guaranteed.  The most dependable way is to check the output of kneighbors,